.github/workflows/ci/pkg/bot/check.go

```diff
@@ -87,38 +86,56 @@ func repoAdminHasApproved(reviews map[string]review) bool {
 // case in which reviews are not dismissed is if they are from GitHub and
 // only update the PR.
 func (c *Bot) checkExternal(ctx context.Context) error {
-	var obsoleteReviews map[string]review
-	var validReviews map[string]review
-
 	pr := c.Environment.Metadata
 	mostRecentReviews, err := c.getMostRecentReviews(ctx)
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	validReviews, obsoleteReviews = splitReviews(pr.HeadSHA, mostRecentReviews)
 	// External contributions require tighter scrutiny than team
 	// contributions. As such reviews from previous pushes must
 	// not carry over to when new changes are added. Github does
 	// not do this automatically, so we must dismiss the reviews
 	// manually if there is a file change.
-	if err = c.hasFileChangeFromLastApprovedReview(ctx); err != nil {
-		err = c.invalidateApprovals(ctx, obsoleteReviews)
+	invalidReviews, err := c.getInvalidReviews(ctx, mostRecentReviews)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	// Delete invalid reviews from map that will be
+	// checked for required approvals.
+	for _, invalidReview := range invalidReviews {
+		delete(mostRecentReviews, invalidReview.name)
+	}
+	if len(invalidReviews) != 0 {
+		err = c.invalidateApprovals(ctx, invalidReviews)
 		if err != nil {
 			return trace.Wrap(err)
 		}
-	} else {
-		// If there are no file changes between current commit and commit where all
-		// reviewers have approved, then all most recent reviews are valid.
-		validReviews = mostRecentReviews
 	}
 	log.Printf("Checking if %v has approvals from the required reviewers %+v", pr.Author, c.Environment.GetReviewersForAuthor(pr.Author))
-	err = hasRequiredApprovals(validReviews, c.Environment.GetReviewersForAuthor(pr.Author))
+	err = hasRequiredApprovals(mostRecentReviews, c.Environment.GetReviewersForAuthor(pr.Author))
 	if err != nil {
 		return trace.Wrap(err)
 	}
 	return nil
 }
 
+// getInvalidReviews gets invalid reviews which are reviews that were submitted before
+// a new non-empty commit was pushed.
+func (c *Bot) getInvalidReviews(ctx context.Context, reviews map[string]review) (map[string]review, error) {
```

Comment:
How about calling them stale or outdated reviews instead of invalid? You wouldn't need the comment explaining what invalid means anymore.